ipv8/peer.py

```diff
@@ -24,11 +24,44 @@ def __init__(self, key, address=("0.0.0.0", 0), intro=True):
             self.key = key
         self.mid = self.key.key_to_hash()
         self.public_key = self.key.pub()
-        self.address = address
+        self.addresses = {}
+        if address is not None:
+            self.addresses[address.__class__] = address
         self.last_response = 0 if intro else time()
         self._lamport_timestamp = 0
         self.pings = deque(maxlen=5)
 
+    @property
+    def address(self) -> Tuple[str, int]:
+        """
+        Deprecated way to retrieve the IPv4 address.
+
+        Use the ``.addresses`` dictionary instead!
+        """
+        from .messaging.interfaces.udp.endpoint import UDPv4Address
+        return self.addresses.get(UDPv4Address, self.addresses.get(tuple, ("0.0.0.0", 0)))
```

Comment:
Are we sure we want to start mixing `tuple`/`UDPv4Address` addresses? I imagine that this could lead to unexpected problems (e.g. in `Network`).

Reply:
I intentionally made `UDPv4Address` a `collections.namedtuple` for this reason, so there are no compatibility issues. Since the last PR we already use `UDPv4Address` from the Endpoint itself and perform this mixing.

We only need this particular Peer logic to deal with Community logic that recreates addresses as `tuple` instances and sets the Peer `address` manually. Thankfully, there are hardly any instances of Communities manually creating addressses and inserting them. Changing the Community logic will be a next PR though.